src/language-server/config/loadConfig.ts

```diff
@@ -25,55 +21,35 @@ const defaultFileNames = [
 ];
 const envFileNames = [".env", ".env.local"];
 
-export const legacyKeyEnvVar = "ENGINE_API_KEY";
 export const keyEnvVar = "APOLLO_KEY";
 
 export interface LoadConfigSettings {
   // the current working directory to start looking for the config
   // config loading only works on node so we default to
   // process.cwd()
-
-  // configPath and fileName are used in conjunction with one another.
-  // i.e. /User/myProj/my.config.js
-  //    => { configPath: '/User/myProj/', configFileName: 'my.config.js' }
-  configPath?: string;
-
-  // if a configFileName is passed in, loadConfig won't accept any other
-  // configs as a fallback.
-  configFileName?: string;
-
-  // used when run by a `Workspace` where we _know_ a config file should be present.
-  requireConfig?: boolean;
-
-  // for CLI usage, we don't _require_ a config file for everything. This allows us to pass in
-  // options to build one at runtime
-  name?: string;
-  type?: "service" | "client";
```

Comment:
The options were adjusted to actual usage in the code base. 
Stuff always passed in isn't optional anymore, stuff that was never passed in is removed.